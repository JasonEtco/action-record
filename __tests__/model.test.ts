import Joi from '@hapi/joi'
import nock from 'nock'
import Model from '../src/model'
import Instance from '../src/instance'

describe('Model', () => {
  let model: Model

  beforeEach(() => {
    model = new Model({
      name: 'user',
      schema: Joi.object({
        login: Joi.string()
      })
    })

    nock('https://api.github.com')
      .get(/^\/search\/issues\?q=/).reply(200, {
        items: [{
          number: 1,
          labels: [{ name: model.name }],
          created_at: '2009-07-12T20:10:41Z',
          body: '```\n{\n  "action_record_id": "some-string",\n  "login": "JasonEtco"\n}\n```'
        }]
      })
  })

  describe('#findOne', () => {
    it('returns the expected record', async () => {
      const record = await model.findOne({ login: 'JasonEtco' }) as Instance

      expect(record).toBeInstanceOf(Instance)
      expect(record.login).toBe('JasonEtco')
      expect(record).toMatchSnapshot()
    })

    it('returns null if no record was found', async () => {
      const record = await model.findOne({
        login: 'nope'
      })

      expect(record).toBeNull()
    })
  })

  describe('#findAll', () => {
    it.todo('returns the expected records')
    it.todo('returns the expected records with a `where` object')
    it.todo('returns null if no records were found')
  })
})