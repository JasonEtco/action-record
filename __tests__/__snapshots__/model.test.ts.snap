// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model #findOne returns the expected record 1`] = `
Instance {
  "action_record_id": "some-string",
  "created_at": "2009-07-12T20:10:41Z",
  "issue_number": 1,
  "modelName": "user",
}
`;
